sympy/solvers/bivariate.py

```diff
@@ -151,28 +151,29 @@ def _lambert(eq, x):
     if X1 != X2:
         return []  # violated assumptions
 
+    # invert the generator X1 so we have x(u)
+    xusolns = solve(X1 - u, x)
+
+    # There are infinitely many branches for LambertW
+    # but only branches for k = -1 and 0 might be real. The k = 0
+    # branch is real and the k = -1 branch is real if the LambertW argumen
+    # in in range [-1/e, 0]. Since `solve` does not return infinite
+    # solutions we will only include the -1 branch if it tests as real.
+    # Otherwise, inclusion of any LambertW in the solution indicates to
+    #  the user that there are imaginary solutions corresponding to
+    # different k values.
+    lambert_real_branches = [-1, 0]
     u = Dummy('rhs')
```

Comment:
This will fail as you are initialising u after using it in `xusolns = solve(X1 - u, x)`

Reply:
done

Reply:
I was just about to commit it. Thanks.